src/coreclr/jit/codegenxarch.cpp

```diff
@@ -4599,6 +4605,24 @@ void CodeGen::genCodeForCmpXchg(GenTreeCmpXchg* tree)
     genProduceReg(tree);
 }
 
+#if defined(TARGET_XARCH) && defined(FEATURE_HW_INTRINSICS)
+void CodeGen::genSIMDDivByZeroCheck(GenTree* oper)
+{
+    noway_assert(oper->OperIs(GT_SIMD_DIV_BY_ZERO_CHECK));
+    GenTreeSIMDDivByZeroChk* divByZeroChk = oper->AsSIMDDivByZeroChk();
+
+    GenTree* simdOp = oper->gtGetOp1();
+
+    genConsumeRegs(simdOp);
+
+    // Current codegen pattern cannot support ZMM registers since vptest only uses XMM/YMM registers
+    noway_assert(simdOp->TypeGet() == TYP_SIMD16 || simdOp->TypeGet() == TYP_SIMD32);
+
+    GetEmitter()->emitIns_R_R(INS_ptest, emitTypeSize(simdOp->TypeGet()), simdOp->GetRegNum(), simdOp->GetRegNum());
```

Comment:
Is this correct as is?

Further down I'm seeing that you're just passing in `op2`. Since `ptest` is setting `ZF` based on `(x & y) == 0`, it means we're logically doing `op2 != 0` here. So this is effectively doing `!EqualsAll(op2, Vector128.Zero)` since the condition is `jne` (jump if `ZF == 0`).

Instead we rather want to be doing `EqualsAny(op2, Vector128.Zero)` which should emit something like:
```asm
vxorps   xmm0, xmm0, xmm0
vpcmpeqd xmm0, xmm0, [op2]
vptest   xmm0, xmm0
jne      ...
```